File: infrastructures/ddd/typescript/src/api/dto/v1/infra/BaseRequestDto.ts
/**
 * Base class for all request DTOs.
 * Request DTOs:
 * - Live in /api/dto/v1/requests/
 * - Suffix with "Request"
 * - Used for HTTP request payloads
 * - Transport only - NO business logic
 * - Validate at edge (API layer)
 * 
 * All request DTOs should extend this class to include common properties.
 */
export abstract class BaseRequestDto {
  // Add common request properties here if needed
  // For example: clientId, requestTimestamp, etc.
}

File: infrastructures/ddd/csharp/src/Api/Dto/V1/Infra/BaseRequestDto.cs
namespace OddlyDdd.Api.Dto.V1.Infra
{
    /// <summary>
    /// Base class for all request DTOs.
    /// Request DTOs are used for HTTP transport only - NO business logic.
    /// They live in /Api/Dto/ and have Request|Response|Dto suffix.
    /// </summary>
    public abstract class BaseRequestDto
    {
        // Base properties that all requests might need can go here
        // Most requests will only add specific properties in derived classes
    }
}
